thriftbp/thrifttest/server.go

```diff
@@ -157,6 +157,7 @@ func (s *Server) Close() error {
 func NewBaseplateServer(
 	store *secrets.Store,
 	processor thrift.TProcessor,
+	suppressor errorsbp.Suppressor,
```

Comment:
I'm not super happy with this function signature change, but this function also makes less sense to convert to a struct arg.

Reply:
I think this probably makes sense to put in `ServerConfig`?

Reply:
That's the tricky part. `ServerConfig` is used by `NewServer`, which is the more customized version of `NewBaseplateServer`. `NewBaseplateServer` actually constructs `ServerConfig` and calls `NewServer` underneath.

In `NewServer`, this is not needed as `NewServer` does not create any middlewares by itself, it just applies all the middlewares passed in.

This is also the reason I said this function makes less sense to have a struct arg, because that new struct will be very confusing alongside `ServerConfig`.

Reply:
OK I changed `ServerConfig` to be used by both `NewServer` and `NewBaseplateServer`. This makes the function signatures much better.

Reply:
Ya, I agree that neither option is ideal.  Looking forward though, I could see a world where we deprecate and remove `NewServer`, and `NewBaseplateServer` is the real focus anyways.